libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +244,108 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
         NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
 }
 
+// send camera field of view status
+void AP_Camera_Backend::send_camera_fov_status(mavlink_channel_t chan)
+{
+    // POI has been requested
+    const AP_AHRS &ahrs = AP::ahrs();
+    Location l;
+    if (!ahrs.get_location(l)) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: vehicle pos unavailable");
+        return;
+    }
+
+    // change vehicle location to ASML
+    l.change_alt_frame(Location::AltFrame::ABSOLUTE);
+
+    // retrieve gimbal attitude
+    auto *mount = AP_Mount::get_singleton();
+    Vector3f att;
+    mount->get_attitude_euler(get_mount_instance(),att[0], att[1], att[2]);
+    if (isnan(att[1]) || isnan(att[2])) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: gimbal attitude unavailable");
+    }
+
+    // project forward from vehicle looking for terrain
+    // start testing at vehicle's location
+    Location test_loc = l, prev_test_loc = l;
+
+#if AP_TERRAIN_AVAILABLE
+    // get terrain altitude (asml) at test_loc
+    bool success = false;
+    AP_Terrain terrain;
+    float terrain_amsl_m, prev_terrain_amsl_m;
+    success = terrain.height_amsl(test_loc, terrain_amsl_m, true);
+    prev_terrain_amsl_m = terrain_amsl_m;
+
+    if (!success) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: failed to get terrain alt");
+        return;
+    }
+
+    float mount_pitch_deg = att[1];
+    float mount_yaw_ef_deg = wrap_180(att[2] + degrees(ahrs.get_yaw()));
+
+    float dist_increment_m;
+    float dist_max;
+    AP_Param::get("TERRAIN_SPACING", dist_increment_m);
```

Comment:
We shouldn't directly grab parameters by name like this because it is quite CPU intensive.  Instead if necessary we should add an accessor to AP_Terrain to get the current spacing.

For POI_DIST_MAX we will probably need to add a new parameter or we could just hardcode it for now to some value like 10km.

Reply:
Currently Hardcoded. But i am not marking this a resolved so later i remember this to be done.